### PR TITLE
fix(leadspace): allow lead space height to grow with content

### PR DESCRIPTION
**Pull request by emyarod**
### Related Ticket(s)

#7155

### Description

This PR updates the lead space to grow in height together with the content. Previously the lead space was constructed with fixed heights in mind, but with the spec update it seems some of the `position` rules throughout the component may no longer be required

content should remain visible and the container grows in height. the image is also scaled accordingly:

image

Confirm that all lead space variants grow as expected according to the updated component spec

Also confirm that #6323 remains resolved as this PR modifies some changes related to that issue (#6336) 

**question for design:**

in the centered lead space with image, the overlay will need to be adjusted if the content exceeds the min height. how should we handle this?

image

### Changelog

**New**

- set object-fit and object-position rules depending on lead space size to match current appearance

**Changed**

- use `min-height` rules
- convert leadspace section to flex container

**Removed**

- obsolete positioning rules on leadspace container

### CODE REVIEW

**Comment by emyarod** [comment]
the bottom padding on the centered leadspace buttons is supposed to be $layout-06 + $layout-07 but it still doesn't seem to be sufficient for the gradient to make things readable

image

**Comment by shixiedesign** [comment]
Hey @emyarod looking great. I think the padding to keep might be different: 

1. Left aligned, Super & Tall: spacing between heading and copy is currently 48px. **Should be 96px or $spacing-12  and 48px $spacing-09 on `md-sm`**

**Current**
image


**Expecting**
image

2. Left aligned, Medium: spacing between heading and copy is currently 16px. **Should be 64px or $spacing-10 (desktop) and 48px $spacing-09 on `md-sm`**

**Current**
image

**Expecting**
image

3. Looks like variable height is not working on Centered lead space right now. Sorry I was wrong. I thought it's already flexible height. If the margin bottom are build in correctly, the entire lead space + gradient should grow, so much so that the transparent part are kept largely within the margin bottom area. 

image

**Comment by emyarod** [comment]
@shixiedesign should the bottom padding be increased more? it's currently `$layout-06 + $layout-07` according to the spec but it seems like it may not be enough

image

File: packages/web-components/src/components/leadspace/leadspace.scss [hunk 1]
```diff
@@ -6,7 +6,8 @@
 //
 
 @import '@carbon/ibmdotcom-styles/scss/globals/vars';
-@import '@carbon/ibmdotcom-styles/scss/components/leadspace/leadspace';
+// @import '@carbon/ibmdotcom-styles/scss/components/leadspace/leadspace';
```
Comment (kennylam) [inline packages/web-components/src/components/leadspace/leadspace.scss:9]: Can remove this.

**Review by kennylam — APPROVED** [review]
Just one small thing, otherwise LGTM thanks @emyarod!

File: packages/web-components/src/components/leadspace/leadspace.scss (hunk 1 above)
Comment (emyarod) [reply packages/web-components/src/components/leadspace/leadspace.scss]: updated, thanks!

**Comment by emyarod** [comment]
@shixiedesign should be ready for review again!

* `super` and `tall` headline bottom margin 96px on `md` and `lg`, 48px on`sm`
* `short` headline bottom margin is now 32px on on `md` and below
* centered with image button group bottom padding should be `layout-06` + `layout-07`

**Comment by shixiedesign** [comment]
Hey @emyarod, I'm still seeing 2 small issues in left aligned:

### Chart of all expected heading margin bottoms, with bug in parentheses:

| Lead space variant | Breakpoint `lg-max` | Breakpoint `sm-md` | Notes |
| ------------------ | --------------------- | --------------------- | --- |
| Super | 96px | 48px | |
| Tall | 96px (bug: currently 48px) | 48px | |
| Medium | 64px | 48px (bug: currently 24px) | |
| Short | 48px | 32px | |
| Center aligned | 48px | 48px | Currently margin top on copy 🤔 | 

**Center aligned lead space** is still showing a few issues: 
1. Margin bottom below buttons are correct, but the background didn't grow to contain it: 
image

2. On `sm-md` breakpoint, the margin bottom below buttons are correct should be 32px:
image

**Comment by emyarod** [comment]
should be good to go now @shixiedesign !

**Review by shixiedesign — APPROVED** [review]
Awesome! Thank you!

**Comment by shixiedesign** [comment]
Noticed 1 regression in Lead space centered, medium & small breakpoint the image is not covering full width of screen anymore. Will wait for another fix. Thanks!

**Comment by emyarod** [comment]
@shixiedesign I believe everything should be resolved now!

**Review by shixiedesign — APPROVED** [review]
Perfecto! 🎉

**Pull request merged**